### ISSUE TITLE
Setup tests are failing in environments without default GCP credentials

### ISSUE THREAD

**Issue by johanbrandhorst** [google/exposure-notifications-server#510]
### TL;DR
Running `go test ./internal/setup` fails in certain user environments. We should either make these tests work in any environment or only run them under certain conditions (flags, build tags?).

**Expected behavior**
Tests should pass regardless of external environments.

**Observed behavior**
Tests failed because of local configuration missing.

### Reproduction
Run the setup tests in a clean environment:
```shell
$ go test ./internal/serverenv/
--- FAIL: TestSetup (0.00s)
    --- FAIL: TestSetup/blobstore (5.24s)
        setup_test.go:166: unable to connect to storage system: storage.NewClient: dialing: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
    --- FAIL: TestSetup/key_manager (5.49s)
        setup_test.go:144: unable to connect to key manager: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
    --- FAIL: TestSetup/secret_manager (6.44s)
        setup_test.go:122: unable to connect to secret manager: secretmanager.NewClient: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
FAIL
FAIL    github.com/google/exposure-notifications-server/internal/setup  6.730s
FAIL
```

**Environment**
- OS: Linux amd64

**Additional information**
I don't have any default GCP credentials configured.

I'd be happy to submit a fix for this, if a resolution is agreed.

**Comment by johanbrandhorst** [comment]
This is still happening on latest master:
```shell
$ git rev-parse HEAD
9ec98717e3e7255529f4b8980d69543c12acf400
$ go test ./...
?       github.com/google/exposure-notifications-server/cmd/cleanup-export      [no test files]
?       github.com/google/exposure-notifications-server/cmd/cleanup-exposure    [no test files]
?       github.com/google/exposure-notifications-server/cmd/export      [no test files]
?       github.com/google/exposure-notifications-server/cmd/exposure    [no test files]
?       github.com/google/exposure-notifications-server/cmd/federationin        [no test files]
?       github.com/google/exposure-notifications-server/cmd/federationout       [no test files]
?       github.com/google/exposure-notifications-server/cmd/generate    [no test files]
?       github.com/google/exposure-notifications-server/cmd/monolith    [no test files]
?       github.com/google/exposure-notifications-server/internal/admin  [no test files]
ok      github.com/google/exposure-notifications-server/internal/admin/authorizedapps   0.078s
ok      github.com/google/exposure-notifications-server/internal/admin/exports  0.065s
?       github.com/google/exposure-notifications-server/internal/admin/healthauthority  [no test files]
?       github.com/google/exposure-notifications-server/internal/admin/index    [no test files]
ok      github.com/google/exposure-notifications-server/internal/admin/siginfo  0.068s
?       github.com/google/exposure-notifications-server/internal/authorizedapp  [no test files]
ok      github.com/google/exposure-notifications-server/internal/authorizedapp/database 9.123s
ok      github.com/google/exposure-notifications-server/internal/authorizedapp/model    0.055s
ok      github.com/google/exposure-notifications-server/internal/base64util     (cached)
ok      github.com/google/exposure-notifications-server/internal/cleanup        9.128s
ok      github.com/google/exposure-notifications-server/internal/database       8.806s
ok      github.com/google/exposure-notifications-server/internal/envconfig      0.044s
ok      github.com/google/exposure-notifications-server/internal/export 0.100s
ok      github.com/google/exposure-notifications-server/internal/export/database        10.645s
ok      github.com/google/exposure-notifications-server/internal/export/model   (cached)
ok      github.com/google/exposure-notifications-server/internal/federationin   0.034s
ok      github.com/google/exposure-notifications-server/internal/federationin/database  8.343s
?       github.com/google/exposure-notifications-server/internal/federationin/model     [no test files]
ok      github.com/google/exposure-notifications-server/internal/federationout  0.088s
ok      github.com/google/exposure-notifications-server/internal/federationout/database 5.526s
?       github.com/google/exposure-notifications-server/internal/flag   [no test files]
?       github.com/google/exposure-notifications-server/internal/generate       [no test files]
ok      github.com/google/exposure-notifications-server/internal/handlers       (cached)
ok      github.com/google/exposure-notifications-server/internal/jsonutil       0.012s
?       github.com/google/exposure-notifications-server/internal/logging        [no test files]
ok      github.com/google/exposure-notifications-server/internal/metrics        (cached)
?       github.com/google/exposure-notifications-server/internal/monolith       [no test files]
?       github.com/google/exposure-notifications-server/internal/observability  [no test files]
?       github.com/google/exposure-notifications-server/internal/pb     [no test files]
?       github.com/google/exposure-notifications-server/internal/pb/export      [no test files]
ok      github.com/google/exposure-notifications-server/internal/publish        65.746s
ok      github.com/google/exposure-notifications-server/internal/publish/database       5.832s
ok      github.com/google/exposure-notifications-server/internal/publish/model  0.017s
?       github.com/google/exposure-notifications-server/internal/retry  [no test files]
ok      github.com/google/exposure-notifications-server/internal/secrets        0.887s
?       github.com/google/exposure-notifications-server/internal/serverenv      [no test files]
2020/06/03 11:39:47 connector: 1591180787: failed to connect to `host=172.17.0.2 user=my-username database=en-server`: dial error (dial tcp 172.17.0.2:5432: connect: connection refused)

2020/06/03 11:39:48 connector: 1591180788: failed to connect to `host=172.17.0.7 user=my-username database=en-server`: dial error (dial tcp 172.17.0.7:5432: connect: connection refused)

2020/06/03 11:39:48 connector: 1591180788: failed to connect to `host=172.17.0.6 user=my-username database=en-server`: dial error (dial tcp 172.17.0.6:5432: connect: connection refused)

2020/06/03 11:39:48 connector: 1591180788: failed to connect to `host=172.17.0.2 user=my-username database=en-server`: dial error (dial tcp 172.17.0.2:5432: connect: connection refused)

2020/06/03 11:39:48 connector: 1591180788: failed to connect to `host=172.17.0.7 user=my-username database=en-server`: dial error (dial tcp 172.17.0.7:5432: connect: connection refused)

2020/06/03 11:39:48 connector: 1591180788: failed to connect to `host=172.17.0.6 user=my-username database=en-server`: dial error (dial tcp 172.17.0.6:5432: connect: connection refused)

2020/06/03 11:39:48 connector: 1591180788: failed to connect to `host=172.17.0.8 user=my-username database=en-server`: dial error (dial tcp 172.17.0.8:5432: connect: connection refused)

2020/06/03 11:39:49 connector: 1591180789: failed to connect to `host=172.17.0.8 user=my-username database=en-server`: dial error (dial tcp 172.17.0.8:5432: connect: connection refused)

2020/06/03 11:39:49 connector: 1591180789: failed to connect to `host=172.17.0.2 user=my-username database=en-server`: dial error (dial tcp 172.17.0.2:5432: connect: connection refused)

2020/06/03 11:39:49 connector: 1591180789: failed to connect to `host=172.17.0.7 user=my-username database=en-server`: dial error (dial tcp 172.17.0.7:5432: connect: connection refused)

2020/06/03 11:39:49 connector: 1591180789: failed to connect to `host=172.17.0.6 user=my-username database=en-server`: dial error (dial tcp 172.17.0.6:5432: connect: connection refused)

2020/06/03 11:39:49 connector: 1591180789: failed to connect to `host=172.17.0.9 user=my-username database=en-server`: dial error (dial tcp 172.17.0.9:5432: connect: connection refused)

2020/06/03 11:39:50 connector: 1591180790: failed to connect to `host=172.17.0.8 user=my-username database=en-server`: dial error (dial tcp 172.17.0.8:5432: connect: connection refused)

2020/06/03 11:39:50 connector: 1591180790: failed to connect to `host=172.17.0.9 user=my-username database=en-server`: dial error (dial tcp 172.17.0.9:5432: connect: connection refused)

2020/06/03 11:39:50 connector: 1591180790: failed to connect to `host=172.17.0.2 user=my-username database=en-server`: dial error (dial tcp 172.17.0.2:5432: connect: connection refused)

2020/06/03 11:39:51 connector: 1591180791: failed to connect to `host=172.17.0.6 user=my-username database=en-server`: server error (FATAL: the database system is starting up (SQLSTATE 57P03))

2020/06/03 11:39:51 connector: 1591180791: failed to connect to `host=172.17.0.9 user=my-username database=en-server`: dial error (dial tcp 172.17.0.9:5432: connect: connection refused)

2020/06/03 11:39:51 connector: 1591180791: failed to connect to `host=172.17.0.8 user=my-username database=en-server`: dial error (dial tcp 172.17.0.8:5432: connect: connection refused)

{"severity":"info","ts":1591180792.1184,"logger":"default","caller":"envconfig/envconfig.go:89","message":"loaded environment"}
{"severity":"info","ts":1591180792.1184728,"logger":"default","caller":"setup/setup.go:104","message":"Effective environment variables: &{database:0xc00043c750}"}
{"severity":"info","ts":1591180792.1184945,"logger":"default","caller":"database/connection.go:40","message":"Creating connection pool."}
{"severity":"info","ts":1591180792.1219497,"logger":"default","caller":"setup/setup.go:142","message":"Effective DB config: &{Name:en-server User:my-username Host:0.0.0.0 Port:32911 SSLMode:disable ConnectionTimeout:0 Password:<hidden> SSLCertPath: SSLKeyPath: SSLRootCertPath: PoolMinConnections: PoolMaxConnections: PoolMaxConnLife:0s PoolMaxConnIdle:0s PoolHealthCheck:0s}"}
{"severity":"info","ts":1591180792.122009,"logger":"default","caller":"setup/setup.go:148","message":"Effective AuthorizedApp config: &{CacheDuration:0s}"}
{"severity":"info","ts":1591180792.1220288,"logger":"default","caller":"database/connection.go:55","message":"Closing connection pool."}
{"severity":"info","ts":1591180792.1220772,"logger":"default","caller":"database/connection.go:55","message":"Closing connection pool."}
2020/06/03 11:39:52 connector: 1591180792: failed to connect to `host=172.17.0.9 user=my-username database=en-server`: dial error (dial tcp 172.17.0.9:5432: connect: connection refused)

2020/06/03 11:39:55 connector: 1591180795: failed to connect to `host=172.17.0.9 user=my-username database=en-server`: dial error (dial tcp 172.17.0.9:5432: connect: connection refused)

{"severity":"info","ts":1591180800.7871847,"logger":"default","caller":"database/connection.go:55","message":"Closing connection pool."}
{"severity":"info","ts":1591180800.85369,"logger":"default","caller":"database/connection.go:55","message":"Closing connection pool."}
{"severity":"info","ts":1591180801.4703436,"logger":"default","caller":"envconfig/envconfig.go:89","message":"loaded environment"}
{"severity":"info","ts":1591180801.4704418,"logger":"default","caller":"setup/setup.go:104","message":"Effective environment variables: &{database:0xc000416340}"}
{"severity":"info","ts":1591180801.4704866,"logger":"default","caller":"database/connection.go:40","message":"Creating connection pool."}
{"severity":"info","ts":1591180801.4826567,"logger":"default","caller":"setup/setup.go:142","message":"Effective DB config: &{Name:en-server User:my-username Host:0.0.0.0 Port:32912 SSLMode:disable ConnectionTimeout:0 Password:<hidden> SSLCertPath: SSLKeyPath: SSLRootCertPath: PoolMinConnections: PoolMaxConnections: PoolMaxConnLife:0s PoolMaxConnIdle:0s PoolHealthCheck:0s}"}
{"severity":"info","ts":1591180801.4827816,"logger":"default","caller":"database/connection.go:55","message":"Closing connection pool."}
{"severity":"info","ts":1591180801.4828541,"logger":"default","caller":"database/connection.go:55","message":"Closing connection pool."}
{"severity":"info","ts":1591180805.246709,"logger":"default","caller":"database/connection.go:55","message":"Closing connection pool."}
--- FAIL: TestSetup (0.00s)
    --- FAIL: TestSetup/default (15.90s)
        setup_test.go:118: unable to connect to secret manager: secretmanager.NewClient: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
    --- FAIL: TestSetup/secret_manager (16.06s)
        setup_test.go:136: unable to connect to secret manager: secretmanager.NewClient: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
    --- FAIL: TestSetup/key_manager (19.80s)
        setup_test.go:158: unable to connect to secret manager: secretmanager.NewClient: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
FAIL
FAIL    github.com/google/exposure-notifications-server/internal/setup  19.838s
ok      github.com/google/exposure-notifications-server/internal/signing        4.275s
ok      github.com/google/exposure-notifications-server/internal/storage        0.058s
?       github.com/google/exposure-notifications-server/internal/util   [no test files]
ok      github.com/google/exposure-notifications-server/internal/verification   19.479s
ok      github.com/google/exposure-notifications-server/internal/verification/database  18.844s
ok      github.com/google/exposure-notifications-server/internal/verification/model     0.056s
ok      github.com/google/exposure-notifications-server/pkg/api/v1alpha1        0.031s
ok      github.com/google/exposure-notifications-server/pkg/verification        0.036s
?       github.com/google/exposure-notifications-server/testing/enclient        [no test files]
?       github.com/google/exposure-notifications-server/tools/admin-console     [no test files]
?       github.com/google/exposure-notifications-server/tools/export-analyzer   [no test files]
?       github.com/google/exposure-notifications-server/tools/export-config     [no test files]
?       github.com/google/exposure-notifications-server/tools/export-generate   [no test files]
?       github.com/google/exposure-notifications-server/tools/exposure-client   [no test files]
?       github.com/google/exposure-notifications-server/tools/federationin-query        [no test files]
?       github.com/google/exposure-notifications-server/tools/federationout-authorization       [no test files]
?       github.com/google/exposure-notifications-server/tools/federationout-test        [no test files]
?       github.com/google/exposure-notifications-server/tools/hexconvert        [no test files]
?       github.com/google/exposure-notifications-server/tools/unwrap-signature  [no test files]
FAIL
